### PR TITLE
feat: migrate Dialog component

### PR DESCRIPTION
**Pull request by Raubzeug**
add Dialog component
add Readme in English

### CODE REVIEW

File: src/components/Dialog/__stories__/Dialog.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
+import React, {useState} from 'react';
+import {Meta, Story} from '@storybook/react/types-6-0';
```
Comment (ValeraS) [inline src/components/Dialog/__stories__/Dialog.stories.tsx:2]:
```suggestion
import {ComponentStory, ComponentMeta} from '@storybook/react';
```

File: src/components/Dialog/__stories__/Dialog.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import {Button} from '../../Button';
+import {Dialog} from '../Dialog';
+import {DialogShowcase} from './DialogShowcase';
+
+export default {
+    title: 'Components/Dialog',
+    component: Dialog,
+} as Meta;
```
Comment (ValeraS) [inline src/components/Dialog/__stories__/Dialog.stories.tsx:10]:
```suggestion
} as ComponentMeta<typeof Dialog>;
```

File: src/components/Dialog/__stories__/Dialog.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+const DefaultTemplate: Story = (args: any) => {
```
Comment (ValeraS) [inline src/components/Dialog/__stories__/Dialog.stories.tsx:12]:
```suggestion
const DefaultTemplate: ComponentStory<typeof Dialog> = (args) => {
```

File: src/components/Dialog/__stories__/Dialog.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+                    textButtonApply="Apply"
+                    textButtonCancel="Cancel"
+                />
+            </Dialog>
+        </>
+    );
+};
+export const Default = DefaultTemplate.bind({});
+
+const ShowcaseTemplate: Story = () => <DialogShowcase />;
```
Comment (ValeraS) [inline src/components/Dialog/__stories__/Dialog.stories.tsx:41]:
```suggestion
const ShowcaseTemplate: ComponentStory<typeof Dialog> = () => <DialogShowcase />;
```

File: src/components/Dialog/__stories__/DialogShowcase.tsx [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+        open: false,
+        showError: false,
+        progress: false,
+    };
+
+    timeout: NodeJS.Timeout | undefined;
+
+    onClose = () => this.setState({open: false, showError: false});
+
+    onClickButtonApply = () => {
```
Comment (ValeraS) [inline src/components/Dialog/__stories__/DialogShowcase.tsx:106]: Let's on* methods rename to handle*

File: src/components/Dialog/ButtonClose/ButtonClose.tsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+const b = block('dialog-btn-close');
+
+export interface ButtonCloseProps {
+    onClose: (
+        event: React.MouseEvent<HTMLElement, MouseEvent>,
+        opts: {isOutsideClick: boolean},
+    ) => void;
+}
+
+export const ButtonClose = ({onClose}: ButtonCloseProps) => {
```
Comment (ValeraS) [inline src/components/Dialog/ButtonClose/ButtonClose.tsx:17]: Let's replace arrow function with `function`

File: src/components/Dialog/DialogBody/DialogBody.tsx [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+import './DialogBody.scss';
+
+const b = block('dialog-body');
+
+export interface DialogBodyProps {
+    children: React.ReactNode;
+    className?: string;
+}
+
+export const DialogBody = (props: DialogBodyProps) => {
```
Comment (ValeraS) [inline src/components/Dialog/DialogBody/DialogBody.tsx:13]: the same

File: src/components/Dialog/DialogHeader/DialogHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,28 @@
...
+const b = block('dialog-header');
+
+export interface DialogHeaderProps {
+    caption?: React.ReactNode;
+    insertBefore?: React.ReactNode;
+    insertAfter?: React.ReactNode;
+    className?: string;
+}
+
+export const DialogHeader = (props: DialogHeaderProps) => {
```
Comment (ValeraS) [inline src/components/Dialog/DialogHeader/DialogHeader.tsx:14]: the same

File: src/components/Dialog/DialogHeader/DialogHeader.tsx [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+        <div className={b(null, className)}>
+            {insertBefore}
+            <div className={b('caption')}>{caption}</div>
+            {insertAfter}
+        </div>
+    );
+};
+
+DialogHeader.defaultProps = {
+    caption: '',
```
Comment (ValeraS) [inline src/components/Dialog/DialogHeader/DialogHeader.tsx:27]: it doesn't seem to do anything. let's remove

**Review by ValeraS — CHANGES_REQUESTED** [review]

File: src/components/Dialog/Dialog.tsx [hunk 10]
```diff
@@ -0,0 +1,101 @@
+import React, {Component} from 'react';
```
Comment (ValeraS) [inline src/components/Dialog/Dialog.tsx:1]:
```suggestion
import React from 'react';
```
Let's get all from React namespace, e.g. `React.Component`

File: src/components/Dialog/ButtonClose/ButtonClose.tsx (hunk 6 above)
Comment (Raubzeug) [reply src/components/Dialog/ButtonClose/ButtonClose.tsx]: Done

File: src/components/Dialog/DialogBody/DialogBody.tsx (hunk 7 above)
Comment (Raubzeug) [reply src/components/Dialog/DialogBody/DialogBody.tsx]: Done

File: src/components/Dialog/DialogHeader/DialogHeader.tsx (hunk 8 above)
Comment (Raubzeug) [reply src/components/Dialog/DialogHeader/DialogHeader.tsx]: Done

File: src/components/Dialog/DialogHeader/DialogHeader.tsx (hunk 9 above)
Comment (Raubzeug) [reply src/components/Dialog/DialogHeader/DialogHeader.tsx]: Fixed

File: src/components/Dialog/Dialog.tsx (hunk 10 above)
Comment (Raubzeug) [reply src/components/Dialog/Dialog.tsx]: Done

**Review by ValeraS — APPROVED** [review]

File: src/components/Dialog/ButtonClose/ButtonClose.tsx [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+import {block} from '../../utils/cn';
+import {Button} from '../../Button';
+import {Icon} from '../../Icon';
+import {CrossIcon} from '../../icons/CrossIcon';
+import './ButtonClose.scss';
+
+const b = block('dialog-btn-close');
+
+export interface ButtonCloseProps {
+    handleClose: (
```
Comment (amje) [inline src/components/Dialog/ButtonClose/ButtonClose.tsx:11]: props should be named `on*`, inner handlers - `handle*`

**Review by amje — CHANGES_REQUESTED** [review]

File: src/components/Dialog/ButtonClose/ButtonClose.tsx (hunk 11 above)
Comment (Raubzeug) [reply src/components/Dialog/ButtonClose/ButtonClose.tsx]: Fixed!

**Review by amje — APPROVED** [review]

**Pull request merged**